modules/python/test/test.py

```diff
@@ -20,7 +25,35 @@
 basedir = os.path.abspath(os.path.dirname(__file__))
 
 def load_tests(loader, tests, pattern):
-    tests.addTests(loader.discover(basedir, pattern=os.environ.get('OPENCV_PYTEST_FILTER', 'test_') + '*.py'))
+    cwd = os.getcwd()
+    config_file = 'opencv_python_tests.cfg'
```

Comment:
In some cases it can be convenient to run tests from directory other than `<build>`.

We could add new  parameter to test app...
```.sh
./opencv/modules/python/test/test.py --repo opencv --test_root opencv --test_root opencv_contrib
```
And discover tests using recursive directory walk based on provided paths.

If we had ability to detect enabled modules list in runtime built into binaries (python at least), we could skip or filter disabled tests dynamically, without extra config files.

Reply:
Main purpose of this patch is enabling testing of opencv_contrib modules.